### PR TITLE
Refactor get_queryset in Tag viewset

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
This PR separates getting the owner of the tags to be returned when listing/retrieving. I think it is more clear this way to determine how to filter the queryset so tags are only available to the corresponding owner.

### CODE REVIEW

File: eox_tagging/api/v1/viewset.py [hunk 1]
```diff
@@ -30,25 +30,52 @@ def get_queryset(self):
...
+            return queryset_union
         except Exception:  # pylint: disable=broad-except
             return queryset.none()
+
+    def __get_request_owner(self, owner_type):
+        """Returns the owners of the tag to filter the queryset."""
+        site = self.__get_site()
+        user = self.__get_user()
+
+        if owner_type is None:
```
Comment (andrey-canon) [inline eox_tagging/api/v1/viewset.py:55]:
```suggestion
        if not owner_type:
```

File: eox_tagging/api/v1/viewset.py [hunk 2]
```diff
@@ -30,25 +30,52 @@ def get_queryset(self):
...
+            return [site, user]
+
+        if owner_type.lower() == "user":
+            return user
+        elif owner_type.lower() == "site":
```
Comment (andrey-canon) [inline eox_tagging/api/v1/viewset.py:60]:
```suggestion
        if owner_type.lower() == "site":
```

add blank space before this line

File: eox_tagging/test/test_viewset.py [hunk 3]
```diff
@@ -63,18 +63,37 @@ def test_retreive_tag(self):
 
     def test_create_tag(self):
```
Comment (andrey-canon) [inline eox_tagging/test/test_viewset.py:64]: This test verifies the response with `"owner_type": "user",` could you add a test with` "owner_type": "site"` ? and could you verify that the owner_type is not None, since the tests show that the line https://github.com/eduNEXT/eox-tagging/blob/72fde4da1a29ea5c8780039192beb2e5ea408702/eox_tagging/api/v1/viewset.py#L58 is not been executed

Comment (mariajgrimaldi) [reply eox_tagging/test/test_viewset.py:64]: Yes! Of course. Let me add those tests

Comment (mariajgrimaldi) [reply eox_tagging/test/test_viewset.py:64]: The owner_type refers to this: /eox_tagging_api/v1/tags/?owner_type=site or /eox_tagging_api/v1/tags/?owner_type=user. If owner_type is none the queryset returned is an union of both. I'll add tests for that.

File: eox_tagging/test/test_viewset.py [hunk 4]
```diff
@@ -97,6 +146,36 @@ def test_filter_by_username(self):
         response = self.client.get(URL)
         self.assertEqual(response.status_code, 200)
 
+    def test_filter_by_owner_user(self):
+        """Used to test getting a tag given its owner of type user."""
+        URL = "/api/v1/tags/?owner_type=user"
```
Comment (andrey-canon) [inline eox_tagging/test/test_viewset.py:151]: reverse and you don't need to create the url with the query params you can do the following https://docs.djangoproject.com/en/3.0/topics/testing/tools/#django.test.Client.get

File: eox_tagging/test/test_viewset.py [hunk 5]
```diff
@@ -97,6 +146,36 @@ def test_filter_by_username(self):
...
+
+        response = self.client.get(URL)
+        data = response.json().get("results")[0]
```
Comment (andrey-canon) [inline eox_tagging/test/test_viewset.py:154]: this is part of the assert section

File: eox_tagging/test/test_viewset.py [hunk 6]
```diff
@@ -97,6 +146,36 @@ def test_filter_by_username(self):
...
+
+        self.assertEqual(data.get("owner_type").lower(), "user")
+
+    def test_filter_by_owner_site(self):
```
Comment (andrey-canon) [inline eox_tagging/test/test_viewset.py:158]: same comments

File: eox_tagging/test/test_viewset.py [hunk 7]
```diff
@@ -97,6 +146,36 @@ def test_filter_by_username(self):
...
+        """Used to test getting a tag given its owner of type user."""
+        URL = "/api/v1/tags/?owner_type=site"
+
+        response = self.client.get(URL)
+        data = response.json().get("results")[0]
+
+        self.assertEqual(data.get("owner_type").lower(), "site")
+
+    def test_filter_by_wrong_owner(self):
```
Comment (andrey-canon) [inline eox_tagging/test/test_viewset.py:167]: same comments

File: eox_tagging/test/test_viewset.py (hunk 3 above)
Comment (mariajgrimaldi) [reply eox_tagging/test/test_viewset.py:77]: Done!

File: eox_tagging/test/test_viewset.py (hunk 4 above)
Comment (mariajgrimaldi) [reply eox_tagging/test/test_viewset.py:151]: fixed

File: eox_tagging/api/v1/viewset.py [hunk 8]
```diff
@@ -30,25 +30,53 @@ def get_queryset(self):
...
+        if not owner_type:
+            return [site, user]
+
+        if owner_type.lower() == "user":
+            return [user]
+
+        if owner_type.lower() == "site":
+            return [site]
+
+        return None
```
Comment (andrey-canon) [inline eox_tagging/api/v1/viewset.py:64]:
```suggestion
        return []
```
in order to return the same kind of data

File: eox_tagging/test/test_viewset.py (hunk 4 above)
Comment (andrey-canon) [reply eox_tagging/test/test_viewset.py:151]: what did you fix ?

Comment (mariajgrimaldi) [reply eox_tagging/test/test_viewset.py:151]: I added the reverse. Sorry I didn't specify

File: eox_tagging/api/v1/viewset.py (hunk 8 above)
Comment (mariajgrimaldi) [reply eox_tagging/api/v1/viewset.py:64]: nice. Much better then

**Comment by andrey-canon** [comment]
@mariajgrimaldi  please rebase and solve conflicts

**Comment by mariajgrimaldi** [comment]
can I merge this? @andrey-canon

**Comment by felipemontoya** [comment]
@andrey-canon I think we are ready here. Can you please look at it and approve if you are satisfied?
